src/elements/Table/TableContext.tsx

```diff
@@ -0,0 +1,38 @@
+import { TableNode, TableCellNode } from '@prezly/story-content-format';
+import React, { useContext, createContext, PropsWithChildren } from 'react';
+
+interface ContextProps {
+    isHeaderCell: (cell: TableCellNode) => boolean;
+}
+
+const TableContext = createContext<ContextProps | null>(null);
+
+interface Props {
+    table: TableNode;
+}
+
+export function TableContextProvider({ table, children }: PropsWithChildren<Props>) {
+    const isHeaderCell = React.useCallback(
+        (cell: TableCellNode) => {
+            return Boolean(
+                (table.header?.includes(TableNode.TableHeader.FIRST_ROW) && isFirstRow(table, cell)) ||
+                    (table.header?.includes(TableNode.TableHeader.FIRST_COLUMN) && isFirstColumn(table, cell)),
+            );
+        },
+        [table],
+    );
```

Comment:
Let's extract the function body into a hooks-free function and then just wrap it with a hook in the end?

```ts
const value = useMemo(() => ({
   isHeaderCell: (cell) => isHeaderCell(table, cell),
}), [table]);

return <TableContext.Provider value={value}>{children}</TableContext.Provider>;
```